webapp/src/dogma/common/components/editor/FileEditor.tsx

```diff
@@ -0,0 +1,178 @@
+import {
+  Box,
+  Button,
+  Flex,
+  Heading,
+  Input,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Spacer,
+  Stack,
+  Tab,
+  TabList,
+  TabPanel,
+  TabPanels,
+  Tabs,
+  Textarea,
+  VStack,
+  useColorMode,
+  useDisclosure,
+} from '@chakra-ui/react';
+import Editor, { DiffEditor, OnMount } from '@monaco-editor/react';
+import { EditModeToggle } from 'dogma/common/components/editor/EditModeToggle';
+import React, { useState, useRef } from 'react';
+import { FcEditImage, FcCancel } from 'react-icons/fc';
+
+export type FileEditorProps = {
+  language: string;
+  originalContent: string;
+};
+
+const FileEditor = ({ language, originalContent }: FileEditorProps) => {
+  const [tabIndex, setTabIndex] = useState(0);
+  const handleTabChange = (index: number) => {
+    setTabIndex(index);
+  };
+  const [fileContent, setFileContent] = useState('');
+  const editorRef = useRef(null);
+  const handleEditorMount: OnMount = (editor) => {
+    editorRef.current = editor;
+    setFileContent(
+      language.toLowerCase() === 'json'
```

Comment:
`language` was lower cased when it was passed.
```suggestion
      language === 'json'
```